pylsp/config/config.py

```diff
@@ -3,15 +3,23 @@
 # pylint: disable=import-outside-toplevel
 
 import logging
+import sys
 from functools import lru_cache
 from typing import List, Mapping, Sequence, Union
 
-import pkg_resources
 import pluggy
 from pluggy._hooks import HookImpl
 
 from pylsp import _utils, hookspecs, uris, PYLSP
 
+# See compatibility note on `group` keyword:
+#   https://docs.python.org/3/library/importlib.metadata.html#entry-points
```

Comment:
> It seems that from `importlib.metadata import entry_points` will work even before 3.10.

Reproduced here from this link, emphasis mine:

> The “selectable” entry points were introduced in `importlib_metadata` 3.6 and Python 3.10. Prior to those changes, `entry_points` accepted no parameters and always returned a dictionary of entry points, keyed by group. For compatibility, if no parameters are passed to `entry_points`, a `SelectableGroups` object is returned, implementing that dict interface. In the future, calling `entry_points` with no parameters will return an `EntryPoints` object. Users **should rely on the selection interface** to retrieve entry points by group.

Before 3.10 stdlib, and older `importlib_metadata`, _all_ of the `entry_points` in `$PREFIX/lib/python-3*` are discovered and parsed without any kind of cache. In a large environment, this can take a number of _seconds_. The venerable `entrypoints` did this a bit better, but is now deprecated, leaving this half-measure until 3.9 EOL to avoid getting bitten by inconsistent return types and performance cliffs.